src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -814,17 +814,116 @@ public void RecordProperties_10()
         }
 
         [Fact]
-        public void EmptyRecord()
+        public void EmptyRecord_01()
         {
             var src = @"
-record C(); ";
+record C();
+
+class Program
+{
+    static void Main()
+    {
+        var x = new C();
+        var y = new C();
+        System.Console.WriteLine(x == y);
+    }
+}
+";
+
+            var verifier = CompileAndVerify(src, expectedOutput: "True");
+            verifier.VerifyDiagnostics();
+
+            verifier.VerifyIL("C..ctor()", @"
+{
+  // Code size        7 (0x7)
+  .maxstack  1
+  IL_0000:  ldarg.0
+  IL_0001:  call       ""object..ctor()""
+  IL_0006:  ret
+}
+");
+
+            var comp = (CSharpCompilation)verifier.Compilation;
+            comp.VerifyEmitDiagnostics();
+
+            var actualMembers = comp.GetMember<NamedTypeSymbol>("C").GetMembers().ToTestDisplayStrings();
+            var expectedMembers = new[]
+            {
+                "C..ctor()",
+                "System.Type C.EqualityContract.get",
+                "System.Type C.EqualityContract { get; }",
+                "System.String C.ToString()",
+                "System.Boolean C." + WellKnownMemberNames.PrintMembersMethodName + "(System.Text.StringBuilder builder)",
+                "System.Boolean C.op_Inequality(C? r1, C? r2)",
+                "System.Boolean C.op_Equality(C? r1, C? r2)",
+                "System.Int32 C.GetHashCode()",
+                "System.Boolean C.Equals(System.Object? obj)",
+                "System.Boolean C.Equals(C? other)",
+                "C C." + WellKnownMemberNames.CloneMethodName + "()",
+                "C..ctor(C original)"
+            };
+            AssertEx.Equal(expectedMembers, actualMembers);
+        }
+
+        [Fact]
+        public void EmptyRecord_02()
+        {
+            var src = @"
+record C()
+{
+    C(int x)
+    {}
+}
+";
 
             var comp = CreateCompilation(src);
             comp.VerifyEmitDiagnostics(
-                // (2,9): error CS8850: A positional record must have a non-empty parameter list
-                // record C();
-                Diagnostic(ErrorCode.ERR_BadRecordDeclaration, "()").WithLocation(2, 9)
-            );
+                // (4,5): error CS8862: A constructor declared in a record with parameters must have 'this' constructor initializer.
```

Comment:
Should the diagnostic message be adjusted, since the record has a parameter list with no parameters in it?

Reply:
> Should the diagnostic message be adjusted, since the record has a parameter list with no parameters in it?

How does this sound? "A constructor declared in a record with parameter list must have 'this' constructor initializer." 


---
In reply to: [490493623](https://github.com/dotnet/roslyn/pull/47805#discussion_r490493623) [](ancestors = 490493623)

Reply:
Sounds good.